internal/tasks/device_render.go

```diff
@@ -0,0 +1,173 @@
+package tasks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	config_latest "github.com/coreos/ignition/v2/config/v3_4"
+	config_latest_types "github.com/coreos/ignition/v2/config/v3_4/types"
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+func DeviceRender(taskManager TaskManager) {
+	for {
+		select {
+		case <-taskManager.ctx.Done():
+			taskManager.log.Info("Received ctx.Done(), stopping")
+			return
+		case resourceRef := <-taskManager.channels[ChannelDeviceRender]:
+			requestID := reqid.NextRequestID()
+			ctx := context.WithValue(context.Background(), middleware.RequestIDKey, requestID)
+			log := log.WithReqIDFromCtx(ctx, taskManager.log)
+			logic := NewDeviceRenderLogic(taskManager, log, taskManager.store, resourceRef)
+			if resourceRef.Op == DeviceRenderOpUpdate {
+				err := logic.RenderDevice(ctx)
+				if err != nil {
+					log.Errorf("failed rendering device %s/%s: %v", resourceRef.OrgID, resourceRef.Name, err)
+				}
+			} else {
+				log.Errorf("DeviceRender called with unexpected kind %s and op %s", resourceRef.Kind, resourceRef.Op)
+			}
+		}
+	}
+}
+
+type DeviceRenderLogic struct {
+	taskManager    TaskManager
+	log            logrus.FieldLogger
+	store          store.Store
+	resourceRef    ResourceReference
+	device         *api.Device
+	renderedConfig *config_latest_types.Config
+}
+
+func NewDeviceRenderLogic(taskManager TaskManager, log logrus.FieldLogger, store store.Store, resourceRef ResourceReference) DeviceRenderLogic {
+	return DeviceRenderLogic{taskManager: taskManager, log: log, store: store, resourceRef: resourceRef}
+}
+
+func (t *DeviceRenderLogic) RenderDevice(ctx context.Context) error {
+	t.log.Infof("Rendering device %s/%s", t.resourceRef.OrgID, t.resourceRef.Name)
```

Comment:
quesiton: should this be debug how chatty do we expect this to be?

Reply:
Once every time a device spec is rendered :)
I'm honestly not sure how chatty is too chatty... I think it's a useful log message.  Perhaps leave it for now and tune in the future if necessary?